src/components/SecretsTable/SecretsTable.test.js

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { fireEvent, render } from 'react-testing-library';
+import SecretsTable from './SecretsTable';
+
+it('SecretsTable renders with no secrets', () => {
+  const props = {
+    initialRows: [
+      {
+        id: 'empty',
+        secret: '-',
+        type: '-',
+        annotations: '-',
+        add: '-'
+      }
+    ],
+    initialHeaders: [
+      { key: 'secret', header: 'Secret' },
+      { key: 'type', header: 'Type' },
+      { key: 'annotations', header: 'Annotations' },
+      { key: 'add', header: 'Add' }
+    ],
+    handleNew: function() {},
+    handleDelete: function() {},
+    loading: false,
+    secretsLength: 0
+  };
+  const { getAllByText, queryByText } = render(<SecretsTable {...props} />);
+  expect(queryByText(/Secret/i)).toBeTruthy();
+  expect(queryByText(/Type/i)).toBeTruthy();
+  expect(queryByText(/Annotations/i)).toBeTruthy();
+  expect(getAllByText('-').length === 4).toBeTruthy();
```

Comment:
`expect(getAllByText('-').length).toEqual(4);`

this will give more useful error message on failure